snyk/cli
# fix: fix parser error in tfplan parser [CFG-2005]

#### What does this PR do?

Fixes an issue in the terraform-plan-parser.ts file that errors when the `expressions` property in a resource is missing, this occurred using the `null-resource` from terraform. This code makes that property optional and handles it. Fixtures are added for the failing tf-plan.json file.

#### How should this be manually tested?

Download the tf-plan-null.json local and run `snyk iac test` on it. The existing CLI will error, the current branch should exit successfully and show no issues.

#### Any background context you want to provide?

Note this does not fix any of the relevant error handling mentioned in the ticket.

#### What are the relevant tickets?

* [CFG-2005]
